### PR TITLE
Document modular canonicalizer

### PR DESCRIPTION
**Pull request by WardBrian**
#### Submission Checklist

- [x] Builds locally 
- [x] New functions marked with `` `r since("VERSION")` ``
- [x] Declare copyright holder and open-source license: see below

#### Summary

Docs portion of https://github.com/stan-dev/stanc3/pull/1058/

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):
Simons Foundation

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: src/stan-users-guide/using-stanc.Rmd [hunk 1]
```diff
@@ -636,60 +640,52 @@ file.
...
-Some of the changes made are:
-
-* Removing unnecessary extra parenthesis
-
-  `y = ((x-1))` $\to$ `y = x - 1`
-
-* Replacing deprecated assignment operators
-
-  `y <- 0` $\to$ `y = 0`
+In addition to automatic formatting, stanc can also "canonicalize" programs but
```
Comment (rok-cesnovar) [inline src/stan-users-guide/using-stanc.Rmd:649]: by instead of but?

File: src/stan-users-guide/using-stanc.Rmd [hunk 2]
```diff
@@ -636,60 +640,52 @@ file.
...
+This can be done by using `stanc --auto-format --canonicalize=...` where `...`
+is a comma-separated list of options. Currently these options are:
 
-  `my_distribution_cdf_log` $\to$ `my_distribution_lcdf`
+- `deprecations`
 
-* Replacing inline conditionals with the ternary if operator
+  Removes deprecated syntax such as `<-` instead of `=`, replacing the
+  deprecated `_log` suffix, replacing deprecated functions with their drop-in
+  replacements, and replaces `get_lp()` with `target()` and
```
Comment (rok-cesnovar) [inline src/stan-users-guide/using-stanc.Rmd:660]: replacing/replaces should be used consistently?

**Comment by WardBrian** [comment]
Good catches @rok-cesnovar

**Review by rok-cesnovar — APPROVED** [review]
Ah, forgot to re-review. Looks great!

**Pull request merged**